test/end-to-end/getRedirectionURL.test.js

```diff
@@ -377,5 +375,152 @@ describe("getRedirectionURL Tests", function () {
+                let page;
+                const exampleEmail = "test@example.com";
+                // Mocha calls cleanup functions even if the test block is skipped, this helps skipping the after block
+                let didSkip = false;
+
+                before(async function () {
+                    let _isPasswordlessSupported = await isPasswordlessSupported();
+                    if (!_isPasswordlessSupported) {
+                        didSkip = true;
+                        this.skip();
+                    }
+
+                    await backendBeforeEach();
+
+                    await fetch(`${TEST_SERVER_BASE_URL}/startst`, {
+                        method: "POST",
+                        headers: [["content-type", "application/json"]],
+                        body: JSON.stringify({
+                            coreConfig: {
+                                passwordless_code_lifetime: 4000,
+                                passwordless_max_code_input_attempts: 3,
+                            },
+                        }),
+                    }).catch(console.error);
+
+                    browser = await puppeteer.launch({
+                        args: ["--no-sandbox", "--disable-setuid-sandbox"],
+                        headless: true,
+                    });
+                    page = await browser.newPage();
+                    // We need to set the localStorage value before the page loads to ensure ST initialises with the correct value
+                    await page.evaluateOnNewDocument(() => {
+                        localStorage.setItem("disableRedirectionAfterSuccessfulSignInUp", "true");
+                        localStorage.removeItem("isNewUserCheck");
+                    });
+                    await clearBrowserCookiesWithoutAffectingConsole(page, []);
+                    await Promise.all([
+                        page.goto(
+                            `${TEST_CLIENT_BASE_URL}/auth?authRecipe=passwordless&passwordlessContactMethodType=EMAIL`
+                        ),
+                        page.waitForNavigation({ waitUntil: "networkidle0" }),
+                    ]);
+                    await setPasswordlessFlowType("EMAIL", "USER_INPUT_CODE");
+                });
+
+                after(async function () {
+                    // Dont cleanup if tests were skipped
+                    if (didSkip) {
+                        return;
+                    }
+                    await browser.close();
+                    await fetch(`${TEST_SERVER_BASE_URL}/after`, {
+                        method: "POST",
+                    }).catch(console.error);
+                    await fetch(`${TEST_SERVER_BASE_URL}/stopst`, {
+                        method: "POST",
+                    }).catch(console.error);
+
+                    return screenshotOnFailure(this, browser);
+                });
+
+                it("should not do any redirection after successful sign up", async function () {
```

Comment:
add for passwordless magic link too. Once the link is consumed, we want to make sure that the user stays on that screen - what does the UI there look like after?